lib/kitchen/instance.rb

```diff
@@ -396,11 +396,23 @@ def setup_action
     # @see Driver::Base#verify
     # @return [self] this instance, used to chain actions
     # @api private
-    def verify_action
+    def verify_action # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
       banner "Verifying #{to_str}..."
       elapsed = action(:verify) do |state|
-        if legacy_ssh_base_driver?
+        # use special handling for busser
```

Comment:
I think we could make this more readable by adding some helper methods.

```
def verifier_busser?(verifier)
   # check if is a Kitchen::Verifier::Vusser
end

def verifier_dummy?(verifier)
  # check if verifier is a Kitchen::Verifier::Dummy
end

def use_legacy_ssh_verifier?(verifier)
  verifier_busser? || verifier_dummy?
end

if legacy_ssh_base_driver? && use_legacy_ssh_verifier?(verifier)
  legacy_ssh_base_verify(state)
elsif legacy_ssh_base_driver?
  verifier.call(driver.legacy_state(state))
else
  verifier.call(state)
end
```

I guess you could classify this as a nitpick, but it reads much cleaner to me and is easier to test.